.github/workflows/code-style.yaml

```diff
@@ -0,0 +1,26 @@
+# Run code style checks on each pull request.
+name: Code style checks
+
+on: [pull_request]
+
+jobs:
+  linting:
+    env:
+      # Configure a constant location for the uv cache
+      UV_CACHE_DIR: /tmp/.uv-cache
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Pixi Environment
+        uses: prefix-dev/setup-pixi@v0.8.1
+        with:
+          pixi-version: v0.25.0
+          cache: true
+          cache-write: ${{ github.event_name == 'push' && github.ref_name == 'main' }}
```

Comment:
Let's add a comment here, since it wasn't obvious to me what this line was doing. 

Only saves caches on main to avoid exceeding GitHub's 10 GB cache limit.
https://github.com/marketplace/actions/setup-pixi#only-save-caches-on-main

Note to self: I should look into this so I can form an informed opinion.

Reply:
pr-tests needs the cache on every push to make sense, I think. Nonetheless, I've added the comment here.